fenicsadapter/fenicsadapter.py

```diff
@@ -1,759 +1,458 @@
-        received from preCICE is ignored. If FEniCS sends data to preCICE, the y-dimension of data and node coordinates
-        is set to zero.
+        return max_dt
 
-        :return: True, if the 2D-3D coupling can be applied
+    def finalize(self):
         """
-        return self._fenics_dimensions == 2 and self._dimensions == 3
-
-    def initialize(self, coupling_subdomain, mesh, read_field, write_field, u_n,
-                   dimension=2, t=0, n=0, dirichlet_boundary=None ):
-        """Initializes remaining attributes. Called once, from the solver.
+        Finalizes the coupling interface. To be called at the end of the simulation
+        """
+        self._interface.finalize()
 
-        :param coupling_subdomain: domain where coupling takes place
-        :param mesh: fenics mesh
-        :param read_field: function applied on the read field
-        :param write_field: function applied on the write field
-        :param u_n: initial data for solution
-        :param dimension: problem dimension
-        :param t: starting time
-        :param n: time step n
-        :param coupling_marker: boundary marker, can be used for coupling, multiple Neumann boundary conditions are applied
+    def get_solver_name(self):
         """
-        self._fenics_dimensions = dimension
+        Returns
+        -------
+        solver_name : string
+            Name of the solver.
+        """
+        return self._config.get_solver_name()
 
-        if self._fenics_dimensions != self._dimensions:
-            logger.warning("fenics_dimension = {} and precice_dimension = {} do not match!".format(self._fenics_dimensions,
-                                                                                            self._dimensions))
-            if self._can_apply_2d_3d_coupling():
-                logger.warning("2D-3D coupling will be applied. Z coordinates of all nodes will be set to zero.")
-            else:
-                raise Exception("fenics_dimension = {}, precice_dimension = {}. "
-                                "No proper treatment for dimensional mismatch is implemented. Aborting!".format(
-                    self._fenics_dimensions,
-                    self._dimensions))
-        if dirichlet_boundary is not None:
-            self._Dirichlet_Boundary=dirichlet_boundary
-
-        self.set_coupling_mesh(mesh, coupling_subdomain)
-        self._set_read_field(read_field)
-        if write_field:
-            self._set_write_field(write_field)
-        self._precice_tau = self._interface.initialize()
-
-        if write_field:
-            if self._interface.is_action_required(precice.action_write_initial_data()):
-                self._write_block_data()
-                self._interface.mark_action_fulfilled(precice.action_write_initial_data())
+    def is_coupling_ongoing(self):
+        """
+        Tag to check if coupling is still going on.
```

Comment:
```suggestion
        Checks if the coupled simulation is still ongoing.
```

Reply:
Note: For functions that are mainly wrappers of existing preCICE API functions, you could provide a reference to the preCICE API function (in the python interface?).

Reply:
I have added the names of the corresponding API functions in their docstring under the section 
```
Notes
-------
```
and also the link to the `python-bindings` file where the API function can be found.

Reply:
I think this approach is good. Note that some functions are still missing the link (see review below).